Update EIP-5496: Change status to Last Call (#5981)
ethereum/EIPs: EIPS/eip-5496.md

The `PrivilegeCloned` event MUST be emitted when `clonePrivilege` is called.

For Compliant contract, it is RECOMMENDED to use [EIP-1271](./eip-1271.md) to validate the signatures.

## Rationale

### Multiple Privileges

[EIP-721](./eip-721.md) only records the ownership and its transfer, the privileges of an NFT are not recorded on-chain. This extension would allow merchants/projects to give out a certain privilege to a specified group of people, and owners of the privileges can manage each one of the privileges independently. This facilitates a great possibility for NFTs to have real usefulness.

### Shareable Privileges

The number of privilege holders is limited by the number of NFTs if privileges are non-shareable. A shareable privilege means the original privilege holder can copy the privilege and give it to others, not transferring his/her own privilege to them. This mechanism greatly enhances the spread of privileges as well as the adoption of NFTs.

### Expire Date Type

> The expiry timestamp of a privilege is a timestamp and stored in `uint256` typed variables.

Reviewer: Why?

Also, does it refer to block number or timetamp?

### Beneficiary of Referrer

> For example, a local pizza shop offers a 30% off Coupon and the owner of the shop encourages their consumers to share the coupon with friends, then the friends can get the coupon. Let's say Tom gets 30% off Coupon from the shop and he shares the coupon with Alice. Alice gets the coupon too and Alice's referrer is Tom. For some certain cases, Tom may get more rewards from the shop. This will help the merchants by spreading the promotion information to more consumers.

Reviewer: ```suggestion
For example, a local pizza shop offers a 30% off Coupon and the owner of the shop encourages their consumers to share the coupon with friends, then the friends can get the coupon. Let's say Tom gets 30% off Coupon from the shop and he shares the coupon with Alice. Alice gets the coupon too and Alice's referrer is Tom. For some certain cases, Tom may get more rewards from the shop. This will help the merchants in spreading the promotion among consumers.
```

### Proposal: NFT Transfer

> If the owner of this standard NFT transfer ownership to another user, there is no impact on "privileges". But errors may be occurred when the owner tries to withdraw the original [EIP-721](./eip-721.md) token from [EIP-5496](./eip-5496.md) via calling function `unwrap()` in our implementation contract if any available privilege is still ongoing. We protect the rights of holders of the privileges by checking the last expire date of the privilege.

Reviewer: ```suggestion
If the owner of the NFT transfers ownership to another user, there is no impact on "privileges". But errors may occur if the owner tries to withdraw the original [EIP-721](./eip-721.md) token from the wrapped NFT through `unwrap()` if any available privileges are still ongoing. We protect the rights of holders of the privileges to check the last expiration date of the privilege.
```
